backtrack/permutation.go

```diff
@@ -0,0 +1,26 @@
+// permutation.go
+// description: generate permutations of a given set
+// details:
+// In mathematics, a permutation of a set is, loosely speaking, an arrangement of its members into a sequence or linear order, or if the set is already ordered, a rearrangement of its elements. The word "permutation" also refers to the act or process of changing the linear order of an ordered set.
+// author: [sixwaaaay](https://github.com/sixwaaaay)
+// see permutation_test.go
+
+package backtrack
+
+func Permutation(nums []int) (set [][]int) {
```

Comment:
The naming doesn't make sense - there is no indication that this is a permutation of a set rather than an elementary permutation. You've written very good code, the only thing I feel could be improved is the naming you have chosen.